src/urbit/app/chess.hoon

```diff
@@ -835,18 +835,66 @@
   ^-  (unit (unit cage))
   ?+  path  (on-peek:default path)
     ::
-    ::  .^(noun %gx /=chess=/game/~1996.2.16..10.00.00..0000/noun)
-    ::  read game info
-    ::  either active or archived
-    [%x %game @ta ~]
+    :: send back a list of chess-games
+    [%x %archive *]
+      ?+  t.t.path  (on-peek:default path)
+      ::
+      ::  =sur -build-file /=chess=/sur/chess/hoon
+      ::  .^((map @dau chess-game:sur) %gx /=chess=/archive/all/noun)
+      ::  .^(json %gx /=chess=/archive/all/json)
+      [%all ~]
+      ``[%chess-archive !>(archive)]
+      ::  XX: include more search criteria encoded in the path
+      ::      ex: opponent    archive/ship
+      ::          date range  archive/@da/@da
+      ::          event/site  archive/@t
+      ::          date area   archive/@da/@dr
+      ::          latest #    archive/last/@ud
+      ::          result      archive/result
+      ::          side      archive/side
+      ::
+      ::      *range, opponent, and side would be better with
+      ::      date range or a limit
+      ::
+      ::  XX:  archive will probably need to be a mop
+      ::  if we want this to be ordered nicely.
+      ::
+      ::  collect only the latest 8 games in archive
+      :: [%last @ud ~]
+      ==
```

Comment:
I think getting the last 8-12 games at a time is probably the correct default number, and the frontend scry should be updated to do that. Looks like it's scrying `/archive/all` right now?

Reply:
yes. I just started with the most basic. searching is pretty complex.